### ISSUE TITLE
Unhelpful/Confusing error handling when creating pages

### ISSUE THREAD

**Issue by davelopez** [galaxyproject/galaxy#17690]
**Describe the bug**
If you omit the `identifier` when creating a page you get a rather "technical" and cryptic error. Also, it looks like the `annotation` is a required value (but it isn't) while the `identifier`, which is required, doesn't look like that.

Detected during 24.0 release testing.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: At least 23.1
Commit: (run `git rev-parse HEAD` if you run this Galaxy server)

**To Reproduce**
1. Go to https://usegalaxy.org/pages/create
2. Enter a name (or not) but keep the ´identifier´ field empty
3. Click `Save`

**Expected behavior**
Instead of the raw error response, the `Indetifier` field should be decorated with a more user-friendly error. Also, the annotation seems to catch more attention than the `Identifier` while it is optional.

**Screenshots**
![Screenshot from 2024-03-13 11-03-25](https://github.com/galaxyproject/galaxy/assets/46503462/a20ae3d6-0437-4d2c-862c-29a8274bf5d1)

**Comment by hexylena** [comment]
the identifier could be generated (or auto suggested) from the title with the history slugging algorithm

**Comment by davelopez** [comment]
I guess the slugging algorithm lives in the backend, I can't remember right now, but it would be helpful to have it directly in the front end so you can generate it as you type the name, and stop auto-generating if you manually edit it.

**Comment by hexylena** [comment]
yeah I mean lowercasing + `s/ /-/g;` is probably sufficient, doesn't need to copy it exactly.
